src/sagas/tokens.js

```diff
@@ -0,0 +1,255 @@
+import {
+  takeEvery,
+  select,
+  delay,
+  call,
+  fork,
+  take,
+  all,
+  put,
+} from 'redux-saga/effects';
+import { metadataApi } from '@hathor/wallet-lib';
+import { channel } from 'redux-saga';
+import { get } from 'lodash';
+import { TxHistory } from '../models';
+import { specificTypeAndPayload } from './helpers';
+import {
+  types,
+  tokenFetchBalanceSuccess,
+  tokenFetchBalanceFailed,
+  tokenFetchHistorySuccess,
+  tokenFetchHistoryFailed,
+} from '../actions';
+
+export const TOKEN_DOWNLOAD_STATUS = {
+  READY: 'ready',
+  FAILED: 'failed',
+  LOADING: 'loading',
+};
+
+const CONCURRENT_FETCH_REQUESTS = 5;
+const METADATA_MAX_RETRIES = 0;
+
+const mapTokenHistory = (element, token) => {
```

Comment:
Moved the method to utils

I agree that it should be a saga, but since this PR is already huge, I suggest doing it in another PR

What do you think?

80cef91ed4fe220690f60f885ae32fb146b9f7c2